src/main/java/core/basesyntax/strategy/handler/DecrementHandler.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.strategy.handler;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.service.OperationHandler;
+
+public class DecrementHandler implements OperationHandler {
+    @Override
+    public void apply(Fruit fruit, int quantity) {
+        Fruit existingFruit = Storage.getFruit(fruit.getName());
+        if (existingFruit == null || existingFruit.getQuantity() < quantity) {
+            throw new RuntimeException("Not enough fruit available.");
+        }
+        existingFruit.subtractQuantity(quantity);
+        Storage.addOrUpdateFruit(existingFruit.getName(), existingFruit);
+    }
```

Comment:
The class `DecrementHandler` should not directly manipulate the `Storage`. Consider following the Single Responsibility Principle and encapsulation principle. Instead, you could have a service that interacts with the `Storage` and this handler should call that service.